src/rpcrawtransaction.cpp

```diff
@@ -604,6 +605,167 @@ UniValue listunspent(const UniValue& params, bool fHelp)
+    {
+        CTxDestination outaddress;
+        int64_t nOutValue = out.tx->vout[out.i].nValue;
+
+        if (!ExtractDestination(out.tx->vout[out.i].scriptPubKey, outaddress)) continue;
+
+        if (CBitcoinAddress(outaddress) == OptimizeAddress)
+            mInputs.insert(std::make_pair(nOutValue, out));
+    }
+
+    CWalletTx wtxNew;
+
+    // For min fee calculation.
+    CTransaction txDummy;
+
+    set<pair<const CWalletTx*,unsigned int>> setCoins;
+
+    unsigned int iInputCount = 0;
+    int64_t nValue = 0;
+
+    // Construct the inputs to the consolidation transaction. Either all of the inputs from above, or 200,
+    // or when the total reaches/exceeds nConsolidateLimit, whichever is more limiting. The map allows us
+    // to elegantly select the UTXO's from the smallest upwards.
+    for (auto const& out : mInputs)
+    {
+        // Increment first so the count is 1 based.
+        ++iInputCount;
+
+        if (fDebug) LogPrintf("INFO: consolidateunspent: input value = %f, confirmations = %" PRId64, ((double) out.first) / (double) COIN, out.second.nDepth);
+
+        setCoins.insert(make_pair(out.second.tx, out.second.i));
+        nValue += out.second.tx->vout[out.second.i].nValue;
+
+        if (iInputCount == nInputNumberLimit || (nValue >= nConsolidateLimit && nConsolidateLimit != 0)) break;
+    }
+
+    // If number of inputs that meet criteria is less than two, then do nothing.
+    if (iInputCount < 2)
+    {
+        result.pushKV("result", true);
+        result.pushKV("UTXOs consolidated", (uint64_t) 0);
+
+        return result;
+    }
+    
+    CReserveKey reservekey(pwalletMain);
+
+
+    // Fee calculation to avoid change.
+
+    // Bytes
+    // --------- The inputs to the tx - The one output.
+    int64_t nBytes = iInputCount * 148 + 34 + 10;
+
+    // Min Fee
+    int64_t nMinFee = txDummy.GetMinFee(1, GMF_SEND, nBytes);
+
+    int64_t nFee = nTransactionFee * (1 + nBytes / 1000);
+
+    int64_t nFeeRequired = max(nMinFee, nFee);
+
+
+    if (pwalletMain->IsLocked())
+    {
+        string strError = _("Error: Wallet locked, unable to create transaction.");
+        LogPrintf("consolidateunspent: %s", strError);
+        return strError;
+    }
+
+    if (fWalletUnlockStakingOnly)
+    {
+        string strError = _("Error: Wallet unlocked for staking only, unable to create transaction.");
+        LogPrintf("consolidateunspent: %s", strError);
+        return strError;
+    }
+
+    vector<pair<CScript, int64_t> > vecSend;
+
+    // Reduce the out value for the transaction by nFeeRequired from the total of the inputs to provide a fee
+    // to the staker. The fee has been calculated so that no change should be produced from the CreateTransaction
+    // call. Just in case, the input address is specified as the return address via coincontrol.
+    vecSend.push_back(std::make_pair(scriptDestPubKey, nValue - nFeeRequired));
+
+    CCoinControl* coinControl = new CCoinControl();
```

Comment:
This will leak unless the receiver takes ownership of the pointer. I'm trying to figure out who owns what but you can either
- Use a stack allocated object
- Use a unique_ptr
- Leave as is if the receiver handles the delete